tool/tctl/common/edit_command_test.go

```diff
@@ -87,3 +88,79 @@ func TestEditGithubConnector(t *testing.T) {
 	assert.Error(t, err, "stale connector was allowed to be updated")
 	require.Error(t, backend.ErrIncorrectRevision, err, "expected an incorrect revision error got %T", err)
 }
+
+func TestEditSAMLConnector(t *testing.T) {
+	modules.SetTestModules(t, &modules.TestModules{
+		TestBuildType: modules.BuildEnterprise,
+		TestFeatures:  modules.Features{SAML: true},
+	})
+	ctx := context.Background()
+	log := utils.NewLoggerForTests()
+	fc, fds := testhelpers.DefaultConfig(t)
+	_ = testhelpers.MakeAndRunTestAuthServer(t, log, fc, fds)
+	rootClient := testhelpers.MakeDefaultAuthClient(t, log, fc)
+	expected, err := types.NewSAMLConnector("saml", types.SAMLConnectorSpecV2{
+		AssertionConsumerService: "test",
+		EntityDescriptor: `<?xml version="1.0" encoding="UTF-8"?>
+    <md:EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" entityID="test">
+      <md:IDPSSODescriptor WantAuthnRequestsSigned="false" protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol">
+        <md:KeyDescriptor use="signing">
+          <ds:KeyInfo xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
+            <ds:X509Data>
+              <ds:X509Certificate></ds:X509Certificate>
+            </ds:X509Data>
+          </ds:KeyInfo>
+        </md:KeyDescriptor>
+        <md:NameIDFormat>urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress</md:NameIDFormat>
+        <md:NameIDFormat>urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified</md:NameIDFormat>
+        <md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="test" />
+        <md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="test" />
+      </md:IDPSSODescriptor>
+    </md:EntityDescriptor>`,
+		Display: "SAML",
+		AttributesToRoles: []types.AttributeMapping{
+			{
+				Name:  "test",
+				Value: "test",
+				Roles: []string{"access"},
+			},
+		},
+	})
+	require.NoError(t, err, "creating initial connector resource")
+
+	created, err := rootClient.CreateSAMLConnector(ctx, expected.(*types.SAMLConnectorV2))
+	require.NoError(t, err, "persisting initial connector resource")
+
+	editor := func(name string) error {
+		f, err := os.Create(name)
+		if err != nil {
+			return trace.Wrap(err, "opening file to edit")
+		}
+
+		expected.SetRevision(created.GetRevision())
+		expected.SetSigningKeyPair(created.GetSigningKeyPair())
+		expected.SetAssertionConsumerService("abcdef")
```

Comment:
IMHO it'd be much easier to understand the test if `expected` had a hardcoded acs that never changes and `created` had its acs changed before we actually call `CreateSAMLConnector`. This way the test could be read top to bottom, whereas now you kinda 

It was hard for me to understand what role `expected` has in this test and what is being updated. I think that's because I expect vars called like `expected` to be hardcoded values that I compare against values returned by running the code under tests. Here `expected` is used for more than that as it's used to create the initial connector as well.

What do you think about changing the test like this?

<details>
<summary>Patch</summary>

```diff
diff --git a/tool/tctl/common/edit_command_test.go b/tool/tctl/common/edit_command_test.go
index bed557fe61..937a701938 100644
--- a/tool/tctl/common/edit_command_test.go
+++ b/tool/tctl/common/edit_command_test.go
@@ -99,8 +99,8 @@ func TestEditSAMLConnector(t *testing.T) {
 	fc, fds := testhelpers.DefaultConfig(t)
 	_ = testhelpers.MakeAndRunTestAuthServer(t, log, fc, fds)
 	rootClient := testhelpers.MakeDefaultAuthClient(t, log, fc)
-	expected, err := types.NewSAMLConnector("saml", types.SAMLConnectorSpecV2{
-		AssertionConsumerService: "test",
+	draft, err := types.NewSAMLConnector("saml", types.SAMLConnectorSpecV2{
+		AssertionConsumerService: "original-acs",
 		EntityDescriptor: `<?xml version="1.0" encoding="UTF-8"?>
     <md:EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" entityID="test">
       <md:IDPSSODescriptor WantAuthnRequestsSigned="false" protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol">
@@ -128,7 +128,7 @@ func TestEditSAMLConnector(t *testing.T) {
 	})
 	require.NoError(t, err, "creating initial connector resource")
 
-	created, err := rootClient.CreateSAMLConnector(ctx, expected.(*types.SAMLConnectorV2))
+	created, err := rootClient.CreateSAMLConnector(ctx, draft.(*types.SAMLConnectorV2))
 	require.NoError(t, err, "persisting initial connector resource")
 
 	editor := func(name string) error {
@@ -137,11 +137,11 @@ func TestEditSAMLConnector(t *testing.T) {
 			return trace.Wrap(err, "opening file to edit")
 		}
 
-		expected.SetRevision(created.GetRevision())
-		expected.SetSigningKeyPair(created.GetSigningKeyPair())
-		expected.SetAssertionConsumerService("abcdef")
+		draft.SetRevision(created.GetRevision())
+		draft.SetSigningKeyPair(created.GetSigningKeyPair())
+		draft.SetAssertionConsumerService("updated-acs")
 
-		collection := &connectorsCollection{saml: []types.SAMLConnector{expected}}
+		collection := &connectorsCollection{saml: []types.SAMLConnector{draft}}
 		return trace.NewAggregate(writeYAML(collection, f), f.Close())
 
 	}
@@ -150,13 +150,13 @@ func TestEditSAMLConnector(t *testing.T) {
 	_, err = runEditCommand(t, fc, []string{"edit", "connector/saml"}, withEditor(editor))
 	require.NoError(t, err, "expected editing saml connector to succeed")
 
-	actual, err := rootClient.GetSAMLConnector(ctx, expected.GetName(), true)
+	actual, err := rootClient.GetSAMLConnector(ctx, draft.GetName(), true)
 	require.NoError(t, err, "retrieving saml connector after edit")
 	require.Empty(t, cmp.Diff(created, actual, cmpopts.IgnoreFields(types.Metadata{}, "ID", "Revision", "Namespace"),
 		cmpopts.IgnoreFields(types.SAMLConnectorSpecV2{}, "AssertionConsumerService"),
 	))
 	require.NotEqual(t, created.GetAssertionConsumerService(), actual.GetAssertionConsumerService(), "acs should have been modified by edit")
-	require.Equal(t, expected.GetAssertionConsumerService(), actual.GetAssertionConsumerService(), "acs should match the retrieved connector")
+	require.Equal(t, "updated-acs", actual.GetAssertionConsumerService(), "the retrieved connector should have updated acs")
 
 	// Try editing the connector a second time this, time the revisions will not match
 	// since the created revision is stale.
```
</details>

Reply:
Makes sense, what about changing just the strings used as acs to "original-acs" and "updated-acs"?